lib/fog/compute/kubevirt/models/vm_data.rb

```diff
@@ -0,0 +1,163 @@
+          return {} if object.nil?
+          disks = []
+          object.each do |d|
+            disk = VmDisk.new
+            disk.name = d[:name]
+            disk.boot_order = d[:bootOrder]
+            disk.volume_name = d[:volumeName]
+
+            if d.keys.include?(:cdrom)
+              disk.type = 'cdrom'
+              disk.bus = d.dig(:cdrom, :bus)
+              disk.readonly = d.dig(:cdrom, :readonly)
+            elsif d.keys.include?(:disk)
+              disk.type = 'disk'
+              disk.bus = d.dig(:disk, :bus)
+              disk.readonly = d.dig(:disk, :readonly)
+            elsif d.keys.include?(:floppy)
+              disk.type = 'floppy'
+              disk.readonly = d.dig(:floppy, :readonly)
+            elsif d.keys.include?(:lun)
+              disk.type = 'lun'
+              disk.readonly = d.dig(:lun, :readonly)
+            end
+            disks << disk
+          end
+
+          disks
+        end
+
+        #
+        # Returns an array of parsed volumes
+        #
+        # @param object [Hash] A hash with raw volumes data.
+        #
+        def parse_volumes(object)
+          return {} if object.nil?
+          volumes = []
+          object.each do |v|
+            volume = VmVolume.new
+            volume.name = v[:name]
+            if v.keys.include?(:registryDisk)
+              volume.type = 'registryDisk'
+              volume.info = v.dig(:registryDisk, :image)
+            elsif v.keys.include?(:persistentVolumeClaim)
+              volume.type = 'persistentVolumeClaim'
+              volume.info = v.dig(:persistentVolumeClaim, :claimName)
+            elsif v.keys.include?(:emptyDisk)
+              volume.type = 'emptyDisk'
+              volume.info = v.dig(:emptyDisk, :capacity)
+            elsif v.keys.include?(:ephemeral)
+              volume.type = 'ephemeral'
+              volume.info = v.dig(:ephemeral, :persistentVolumeClaim, :claimName)
+            elsif v.keys.include?(:cloudInitNoCloud)
+              volume.type = 'cloudInitNoCloud'
+              volume.info = v.dig(:cloudInitNoCloud, :userDataBase64)
+            elsif v.keys.include?(:hostDisk)
+              volume.type = 'hostDisk'
+              volume.info = v.dig(:hostDisk, :path)
+            elsif v.keys.include?(:secret)
+              volume.type = 'secret'
+              volume.info = v.dig(:secret, :secretName)
+            elsif v.keys.include?(:dataVolume)
+              volume.type = 'dataVolume'
+              volume.info = v.dig(:dataVolume, :name)
+            elsif v.keys.include?(:serviceAccount)
+              volume.type = 'serviceAccount'
+              volume.info = v.dig(:serviceAccount, :serviceAccountName)
+            elsif v.keys.include?(:configMap)
+              volume.type = 'configMap'
+              volume.info = v.dig(:configMap, :name)
+            end
+            volumes << volume
+          end
+
+          volumes
+        end
+
+        class VmNic
```

Comment:
It seems like we redefined those classes one more time. Previous definition is [here](https://github.com/fog/fog-kubevirt/pull/39/files#diff-ec5834173b387b1cf1116d4079b963a3R82).